### PR TITLE
Portalnetwork test coverage

### PR DESCRIPTION
**Pull request by ScottyPoi**
Adds to integration test.

Earmarked: 
 - Reduced logging from uTP: `ContentReader` and `ContentWriter`
 - Small fixes to `PortalNetworkUtp`
 - Fix bug in `ContentWriter` that skipped first data chunk
 - Tweaks to the way history protocol handles snapshots.
 - Sets History gossip queue pulse to 26 (max size for offer)

Integration test currently covers:

- Ping/Pong
- Find Content (Header)
  - Header
  - Block
  - Accumulator Snapshot
  - Epoch Accumulator
- Offer/Accept
  - Headers and Blocks
- Content Gossip
- eth_getBlockByHash
- eth_getBlockByNumber
  - For a block within the current epoch of an accumulator snapshot.
  - For a block from a historical_epoch

### CODE REVIEW

**Comment by ScottyPoi** [comment]
Integration test building revealed a bug in uTP protocol during FOUNDCONTENT requests where ContentWriter would skip the first chunk of data when streaming data packets.

Problem involved when and how seqNr's are set, and read by the writer / socket / packet creator.  
Though I did find the problem and fix it, I feel that there may be a better way to approach this part of the uTP protocol where we increment seqNr's and ackNr's.

Will think on this and propose something in an issue.

File: packages/portalnetwork/src/subprotocols/history/history.ts [hunk 1]
```diff
@@ -430,7 +434,7 @@ export class HistoryProtocol extends BaseProtocol {
     ) {
       // Gossip new content to network (except header accumulators)
       this.gossipQueue.push([hashKey, contentType])
-      if (this.gossipQueue.length >= 5) {
+      if (this.gossipQueue.length >= 26) {
```
[the remark this answers is not in the document]
Comment (ScottyPoi) [reply packages/portalnetwork/src/subprotocols/history/history.ts:437]: we could create a way to configure this number to anything between 1 and 26, or even dynamically change the number based on traffic or something.
for now, I set it to MAX (26) for the sake of performance.  Larger offers = fewer handshakes.

**Review by ScottyPoi — COMMENTED** [review]
@acolytec3, I am still writing integration tests, but I recommend merging this whenever you get a chance to look it over, as it does include some bug fixes as well.

**Comment by ScottyPoi** [comment]
added a test for eth_getBlockByNumber 
constraints of the test setup restrict it to blocks within the current epoch of a received accumulator snapshot, for now.

File: packages/portalnetwork/src/subprotocols/history/history.ts [hunk 2]
```diff
@@ -277,41 +279,51 @@ export class HistoryProtocol extends BaseProtocol {
...
-        epoch = EpochAccumulator.deserialize(encodedEpoch)
+        const block = await this.getBlockByHash(blockHash, includeTransactions)
+        return block
+      } catch (err) {
+        this.logger(`getBlockByNumber error: ${(err as any).message}`)
       }
-
-      blockHash = toHexString(epoch[blockIndex].blockHash)
+    } else {
+      throw new Error(`Blockhash index lookup not implemented`)
```
Comment (acolytec3) [inline packages/portalnetwork/src/subprotocols/history/history.ts:293]: I'm not sure I like the idea of throwing here.  It's not an error if we don't find the block or if the number is too high for the accumulator. I think we should just return undefined.

**Review by acolytec3 — CHANGES_REQUESTED** [review]
This generally looks good but I don't love the idea of `getBlockByNumber` throwing.  Would prefer to handle as just returning undefined.  Also, can you remove commented out code that is duplicated by your fixes? I'm going to do some more indepth testing from browser client this afternoon but looks great overall.

**Comment by ScottyPoi** [comment]
Made some edits to enable a test for eth_getBlockByNumber with a historical epoch lookup involved.  
There's some weirdness going on in PortalNetworkUtp._handleFinPacket()  when the content is an epoch accumulator, where the contentKey stored in the utp request is undefined.  Since it is possible to rehash the accumulator to get the proper contentID, the methods work anyway, but it is still an unsolved mystery to me.

**Review by acolytec3 — APPROVED** [review]

**Pull request merged**